tests/rbd_mirror/test_rbd_snapshot_mirroring_with_upgrade.py

```diff
@@ -0,0 +1,245 @@
+from copy import deepcopy
+
+from ceph.parallel import parallel
+from ceph.rbd.initial_config import initial_mirror_config, random_string
+from ceph.rbd.utils import getdict
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.krbd_io_handler import krbd_io_handler
+from ceph.rbd.workflows.parallel import execute
+from ceph.rbd.workflows.snap_scheduling import verify_snapshot_schedule
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run_io_verify_snap_schedule(**kw):
+    """
+    Run IOs on the given image and verify snapshot schedule
+    """
+    pool_type = kw.get("pool_type")
+    rbd = kw.get("rbd")
+    client = kw.get("client")
+    config = deepcopy(kw.get("config").get(pool_type))
+    for pool, pool_config in getdict(config).items():
+        multi_image_config = getdict(pool_config)
+        multi_image_config.pop("test_config", {})
+        for image, image_config in multi_image_config.items():
+            image_spec = f"{pool}/{image}"
+
+            io_size = image_config.get(
+                "io_size", int(int(image_config["size"][:-1]) / 3)
+            )
+            io_config = {
+                "rbd_obj": rbd,
+                "client": client,
+                "size": image_config["size"],
+                "do_not_create_image": True,
```

Comment:
why this config here `do_not_create_image` as images are already created?

Reply:
exactly since image is already created we specify this so that krbd_io_handler doesn't try to create image again.